modules/aws-databricks-unity-catalog/main.tf

```diff
@@ -8,23 +13,11 @@ resource "databricks_metastore" "this" {
 
 resource "databricks_metastore_data_access" "this" {
   metastore_id = databricks_metastore.this.id
-  name         = aws_iam_role.metastore_data_access.name
+  name         = local.iam_role_name
   aws_iam_role {
-    role_arn = aws_iam_role.metastore_data_access.arn
+    role_arn = local.iam_role_arn
   }
   is_default = true
-  depends_on = [
-    resource.time_sleep.wait_role_creation
-  ]
-}
-
-# Sleeping for 20s to wait for the workspace to enable identity federation
-resource "time_sleep" "wait_role_creation" {
-  depends_on = [
-    resource.aws_iam_role.metastore_data_access,
-    resource.databricks_metastore.this
-  ]
-  create_duration = "20s"
 }
```

Comment:
Should we use built-in data source instead?

Reply:
Which are you talking about?

Reply:
about IAM role for metastore access... @flaviomalavazi do you remember why we needed this delay?

Reply:
@mgyucht answer from @flaviomalavazi 

> We needed the delay for the identity federation to be enabled at the workspace (it depended on the workspace having a metastore assigned to it)

Reply:
Interesting. Applying this template as is does work, so maybe it is OK for now? I could also add another sleep for 20s after the IAM role is created to preserve the original behavior, let me know what you think.

Reply:
Added it back.